### ISSUE TITLE
Stuck in loading state for several seconds after requesting from a Bliz server

### ISSUE THREAD

**Issue by cloudrac3r** [snoe/deedum#60]
Hi, I really like this browser!

I've been having an issue with when I'm connecting to my site gemini://cadence.moe/ . The page appears just fine. But the url bar remains green (in the loading state) for several more seconds. I cannot click any links on the page until it decides to stop and turn orange. This makes it super frustrating to navigate.

This issue happens every time I load a page, so you should be able to reproduce the issue yourself really easily.

The logs say that the connection timed out after 10 seconds - but this is strange, and it's especially strange since the page still renders correctly.

I haven't had any problems with other clients, including using the openssl command line; `printf '/\r\n' | openssl s_client -connect cadence.moe:1965 -quiet` works as expected and does not pause.

If this is an issue in Bliz not doing connections properly, let me know and I can try to fix it on my end! It's weird that it's fine in other gemini browsers though.

Thank you!

**Comment by snoe** [comment]
@cloudrac3r This happens with servers that do not close the ssl connection, many libraries will need to do that twice, once for the underlying socket and once for the ssl wrapper. I believe the gemini spec is more explicit about this now but I have to admit I haven't been following development as closely.

Here's an investigation I did for the jetforce server:
https://github.com/michael-lazar/jetforce/issues/32

**Comment by cloudrac3r** [comment]
I read the jetforce issue report, I understood some of it, but I'm not 100% sure which behaviour is the actual behaviour and which behaviour is the expected behaviour.

What I can tell you is that I tracked down the behaviour that bliz is currently exhibiting. Bliz uses this command line to do TLS:

`ncat -k -l $gemini_port -e '/usr/bin/env fish src/cgi.fish' --ssl --ssl-cert certs/cert.pem --ssl-key certs/key.pem`

It uses the ncat binary to call another file with stdin/stdout to be sending and receiving data.

I think the code path is here: https://github.com/nmap/nmap/blob/df33da47228e3f32f9a332f7db0a0a4f2f14084d/ncat/ncat_listen.c#L594

This calls this SSL_shutdown function from the openssl library, which is well-documented: https://www.openssl.org/docs/man1.0.2/man3/SSL_shutdown.html

I'm unable to see how to proceed in debugging this, though.

**Comment by snoe** [comment]
@cloudrac3r not sure, you can see I use `ncat` to test gemini as well https://github.com/snoe/deedum/blob/master/server#L8 `-C` and `-c` vs `-e`  seem to be the only difference. It'd be interesting to see if if it's `cgi.fish` or the `-C`  or maybe the os causing the issue. 

Can you start by trying `server` in this repo (passphrase is "test") and see if you get the same behaviour on your end? 
Next, try your certs, but my `-c` and `-C` command.
Next, try dropping the `-C`
Next, put the `-C` back and use your `-e`


My version of ncat, on os x
```
$ ncat -h
Ncat 7.91 ( https://nmap.org/ncat )
```

**Comment by snoe** [comment]
Something else I notice when I use `server` I get this in this in the log 
https://github.com/snoe/deedum/blob/master/lib/net.dart#L245
But I don't see that at your site. (You can see Logs in the app from the drop down menu)

**Comment by cloudrac3r** [comment]
Oops, I was accidentally very misleading. I tried your ncat on my computer and it worked. And then I tried Bliz on my computer and it worked. So the issue wasn't in ncat - the issue was in Bliz on my cadence.moe server falling back to actually using socat because ncat wasn't available!

So it seems like a problem in socat. ncat works just fine with Bliz.

ncat has always been the preferred option for Bliz anyway, so installing ncat is the correct solution. I'll go and note this in the Bliz readme - thanks for your patience!
